packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -64,24 +72,39 @@ class AttachCommand extends FlutterCommand {
     return null;
   }
 
+  @override
+  Future<Null> validateCommand() async {
+    super.validateCommand();
+    if (await findTargetDevice() == null)
+      throwToolExit(null);
+    observatoryPort;
+  }
+
   @override
   Future<Null> runCommand() async {
     Cache.releaseLockEarly();
 
     await _validateArguments();
 
     final Device device = await findTargetDevice();
-    if (device == null)
-      throwToolExit(null);
     final int devicePort = observatoryPort;
+
+    final Daemon daemon = argResults['machine']
+      ? new Daemon(
+            stdinCommandStream, stdoutCommandResponse,
+            notifyingLogger: new NotifyingLogger(), logToStdout: true)
```

Comment:
Trailing commas would help improve formatting here and several places below, cf https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation